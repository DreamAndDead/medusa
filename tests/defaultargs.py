def hello(name, age=20, nickname="", *args):
    print("Hello, my name is " + name + " and I'm " + str(age))
    print("My nickname is " + nickname)

hello("John", 12, "antikiller")
hello("Josh", 45)
hello("Jane")
