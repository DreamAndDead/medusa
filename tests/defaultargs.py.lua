dofile('./pythonlua/luainit.lua')

local function hello(name, age, nickname, ...)
    age = age or 20
    nickname = nickname or ""
    local args = list {...}
    print(((("Hello, my name is " + name) + " and I'm ") + str(age)))
    print(("My nickname is " + nickname))
end
hello("John", 12, "antikiller")
hello("Josh", 45)
hello("Jane")