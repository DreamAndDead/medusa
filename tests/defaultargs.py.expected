Hello, my name is John and I'm 12
My nickname is antikiller
Hello, my name is Josh and I'm 45
My nickname is 
Hello, my name is Jane and I'm 20
My nickname is 
